plugins/modules/source_control/gitlab/gitlab_runner.py

```diff
@@ -248,7 +258,7 @@ def create_runner(self, arguments):
             return True
 
         try:
-            runner = self._runners_endpoint.create(arguments)
+            runner = self._gitlab.runners.create(arguments)
```

Comment:
Does this make sense? How does GitLab know whether this runner belongs to a group, a project, or is a global runner?

Reply:
Gitlab manages its runners as an absolute object, which can be linked to a group/project.
The project/group concerned by the runner declaration is guessed by Gitlab thanks to the `token` parameter, which is unique for an instance/group/project (within a Gitlab instance, each group and each project has its own registration token).

[Gitlab API Resources docs][1] describes the CRUD actions on runners as root resources (/runners/:id) and not group or project-related resources (/project/:project_id/runners/:id).

`_gitlab.runners` object is a wrapper of Gitlab instance runners endpoint (/runners/:id), while `_runners_endpoint` object can represent instance, project or group-related runner API endpoint, depending on `owned`, `group` and `project` parameters values.
Thus, the `_gitlab.runners` wrapper must be used for registration, update, details reading or deletion.
The `_runners_endpoint` wrapper will still be used for browsing declared instance/group/project runners list.

[1]: https://docs.gitlab.com/ee/api/runners.html

Reply:
Bugfix addressed in #3965